spec/integration/parser/compiler_spec.rb

```diff
@@ -522,4 +522,31 @@ class xport_test {
       expect(catalog).to have_resource("Notify[nbr2]").with_parameter(:message, 'overridden')
     end
   end
+
+
+  context "when converting catalog to resource" do
+    it "the same environment is used for compilation as for transformation to resource form" do
+        Puppet[:code] = <<-MANIFEST
+          notify { 'dummy':
+          }
+        MANIFEST
+
+        # Create an instance of a mocha state machine that checks that the environment is
+        # 'production'
+        in_production_state = Class.new do
+
+          def active?
+            Puppet.lookup(:current_environment).name == :production
+          end
+
+          def to_s
+            # displayed if the expectation about this state fails
+            "in production environment"
+          end
+        end.new
```

Comment:
@hlindberg how about https://github.com/joshcooper/puppet/commit/ed9ee3aaf9f4b8095900e6f051123ad1f975430d. Also, btw it looks like one of your commits as PUP-3775 instead of PUP-3755